fastlane/Fastfile

```diff
@@ -584,28 +584,59 @@ end
-        errors << "Use <em>#{tool}<em> instead of <code>#{tool}</code> to mention a tool in the docs in '#{path}'" if content.include?("<code>#{tool}</code>")
-        if content.include?("_#{tool.to_s.capitalize}_") || content.include?("`#{tool.to_s.capitalize}`")
-          errors << "fastlane tools have to be formatted in lower case: #{tool} in '#{path}'"
-        end
-      end
+      errors += find_tool_name_formatting_errors(content: content, path: path)
     end
   end
   errors.each { |a| UI.error(a) }
   UI.user_error!("Invalid formatting of one of the fastlane tools") unless errors.empty?
   UI.success("✅  fastlane tools formatting is correct")
 end
 
+private_lane :test_tool_name_formatting_ensurance_lane do
+  content = File.read("spec/fixtures/fastfiles/tool_name_formatting.txt")
+  errors = find_tool_name_formatting_errors(content: content, path: "CONTRIBUTING.md")
+  errors.select! { |e| !e.end_with?("❌") }
+  errors.each do |e|
+    # Lines that end with "❌" are expected to be errors
+    UI.error(e)
+  end
+  UI.user_error!("Tool name formatting tool failed its unit tests") unless errors.empty?
+  UI.success("✅  Tool name formatting tool passed its unit tests")
+end
+
+def find_tool_name_formatting_errors(content:, path:)
+  errors = []
+  Fastlane::TOOLS.each do |tool|
+    lines = content.split("\n")
+    lines.each_with_index do |line, index|
+      line_number = index + 1
+      # Ignore checks if line starts with "^\s*- ❌" (i.e. if it's a line that's already been marked as incorrect)
+      # and if the file is CONTRIBUTING.md (i.e. if it's the branding guidelines section)
+      next if line =~ /^\s*- ❌/ && path == "CONTRIBUTING.md"
+      errors << "Use _#{tool}_ instead of `#{tool}` to mention a tool in the docs in '#{path}:#{line_number}': #{line}" if line.include?("`#{tool}`")
+      errors << "Use _#{tool}_ instead of `_#{tool}_` to mention a tool in the docs in '#{path}:#{line_number}': #{line}" if line.include?("`_#{tool}_`")
+      errors << "Use [_#{tool}_] instead of [#{tool}] to mention a tool in the docs in '#{path}:#{line_number}': #{line}" if line.include?("[#{tool}]")
+      errors << "Use _#{tool}_ instead of **#{tool}** to mention a tool in the docs in '#{path}:#{line_number}': #{line}" if line.include?("**#{tool}**")
+      errors << "Use <em>#{tool}<em> instead of <code>#{tool}</code> to mention a tool in the docs in '#{path}:#{line_number}': #{line}" if line.include?("<code>#{tool}</code>")
+      errors << "Use <em>#{tool}<em> or _#{tool}_ instead of <ins>#{tool}</ins> to mention a tool in the docs in '#{path}:#{line_number}': #{line}" if line.include?("<ins>#{tool}</ins>")
+
+      line.scan(/([A-Z_]*)([_ `"])(#{Regexp.escape(tool.to_s)})\2([A-Z_]*)/i) do |prefix, delim, tool_name, suffix|
+        wrong_case = tool_name != tool.to_s.downcase
+        looks_like_an_env_var = (tool_name == tool_name.upcase) && delim == '_' && (!prefix.empty? || !suffix.emtpy?)
```

Comment:
Typo (my bad)
```suggestion
        looks_like_an_env_var = (tool_name == tool_name.upcase) && delim == '_' && (!prefix.empty? || !suffix.empty?)
```

Reply:
The fact that your pseudo-unit-test didn't crash due to this typo means that this case wasn't covered by your fixture btw.

I'd suggest some red-green-refactoring of the fixture (ie first change the fixture to include a case of an env var which _starts_ with a tool name, then run the test and confirm it goes red (well… or crashes, in this case), and _only then_ fix the typo in the code and confirm the test goes green); that way you'll be sure to have covered the use case properly.

Reply:
Great catch 👀 
I did the red-green-refactoring strategy and fixed it 💪